### PR TITLE
fix an issue #387(Add Non-UM User: include found user's name on the screen)

### LINKED ISSUE #387: Add Non-UM User: include found user's name on the screen
When adding a non-UM user with the one at a time option and the user already exists, present that user's first and last name on the scree along with the "user found" message. This will help the tool user to make sure the external email address they entered was the right one and aligns with how the old CCM tool worked.

### PR DESCRIPTION
**Pull request by show981111**
Resolves #387 

Added the interface, CanvasUserCondensed, which contains name and loginId 
Returns this interface at userExists functions.

### CODE REVIEW

File: ccm_web/server/src/canvas/canvas.interfaces.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { CanvasOAuthReturnQueryDto } from './dtos/canvas.oauth.query.dto'
 import { hasKeys } from '../typeUtils'
+import { string } from 'yup'
```
Comment (ssciolla) [inline ccm_web/server/src/canvas/canvas.interfaces.ts:3]: Remove?

File: ccm_web/client/src/api.ts [hunk 2]
```diff
@@ -156,9 +157,10 @@ export const unmergeSections = async (sectionsToUnmerge: CanvasCourseSection[]):
   return await resp.json()
 }
 
-export const checkIfUserExists = async (loginId: string): Promise<boolean> => {
+export const checkIfUserExists = async (loginId: string): Promise<CanvasUserCondensed | false> => {
   const request = getGet()
   const resp = await fetch(`/api/admin/user/${loginId}`, request)
+  console.log(resp)
```
Comment (pushyamig) [inline ccm_web/client/src/api.ts:163]: Remove?

**Comment by pushyamig** [comment]
This change seems to be straight forward. @show981111 thanks for making this change. @ssciolla will be the approver.

File: ccm_web/server/src/canvas/canvas.interfaces.ts [hunk 3]
```diff
@@ -223,6 +223,11 @@ export interface CanvasUniqueIdErrorsBody extends CanvasErrorsBody {
   }
 }
 
+export interface CanvasUserCondensed {
+  loginId: string
+  name: string
+}
```
Comment (ssciolla) [inline ccm_web/server/src/canvas/canvas.interfaces.ts:229]: Couple things here:
1) Let's move this so it's next to the other `CanvasUser` interface.
2) This interface could be based on `CanvasUser`, or vice versa. I guess I see this as a limited subset of the existing object, so we could use [`Pick`](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) here, and not redefine `login_id` as `loginId`.
```
type CondensedCanvasUser = Pick<CanvasUser, 'login_id' | 'name'>
```

**Review by ssciolla — COMMENTED** [review]
Looking good, @show981111. I added a couple comments and suggestions that I'd like to see addressed. Let me know if you have any questions. I'm going to run it locally now and test.

File: ccm_web/client/src/components/UserEnrollmentForm.tsx [hunk 4]
```diff
@@ -178,7 +178,7 @@ export default function UserEnrollmentForm (props: UserEnrollmentFormProps): JSX
         Enter the user&apos;s non-UM email address, and click &quot;Search&quot; to see if they are in Canvas. (Required)
       </Typography>
       {
-        showIncompleteAlerts && userExists === undefined &&
+        showIncompleteAlerts && (userInfo === undefined) &&
```
Comment (ssciolla) [inline ccm_web/client/src/components/UserEnrollmentForm.tsx:181]: Just FYI, parentheses aren't really necessary here, but not a big deal, they can help readability.

File: ccm_web/client/src/models/canvas.ts [hunk 5]
```diff
@@ -140,3 +140,8 @@ export const injectCourseName = (sections: CanvasCourseSection[], courseName: st
 export function sortSections<T extends CanvasCourseSection> (sections: T[]): T[] {
   return sections.sort((a, b) => { return a.name.localeCompare(b.name, undefined, { numeric: true }) })
 }
+
+export interface CanvasUserCondensed {
+  loginId: string
```
Comment (ssciolla) [inline ccm_web/client/src/models/canvas.ts:145]: I think this still needs to be updated to be `login_id`? I guess it didn't come up as an issue cause you aren't using the property, but we should probably fix it.

File: ccm_web/client/src/components/UserEnrollmentForm.tsx (hunk 4 above)
Comment (ssciolla) [reply ccm_web/client/src/components/UserEnrollmentForm.tsx:181]: Fine to leave this as is.

File: ccm_web/server/src/canvas/canvas.interfaces.ts [hunk 6]
```diff
@@ -223,6 +223,8 @@ export interface CanvasUniqueIdErrorsBody extends CanvasErrorsBody {
   }
 }
 
+export type CanvasUserCondensed = Pick<CanvasUser, 'login_id' | 'name'>
```
Comment (ssciolla) [inline ccm_web/server/src/canvas/canvas.interfaces.ts:226]: Please move this so it is next to the other `CanvasUser` type/interface.

**Pull request merged**